Sources/MapboxCoreNavigation/BillingHandler.swift

```diff
@@ -0,0 +1,393 @@
+// IMPORTANT: Tampering with any file that contains billing code is a violation of our ToS and will result in enforcement of the
+// penalties stipulated in the ToS.
+
+import Foundation
+import MapboxDirections
+import os.log
+@_implementationOnly import MapboxCommon_Private
+
+/// BillingService from MapboxCommon
+private typealias BillingServiceNative = MapboxCommon_Private.BillingService
+/// BillingServiceError from MapboxCommon
+private typealias BillingServiceErrorNative = MapboxCommon_Private.BillingServiceError
+
+/// Swift variant of `BillingServiceErrorNative`
+enum BillingServiceError: Error {
+    /// Unknown error from Billing Service
+    case unknown
+
+    /// Provided SKU ID is invalid.
+    case invalidSkuId
+    /// The request failed because the access token is invalid.
+    case tokenValidationFailed
+    /// The resume failed because the session doesn't exist or invalid.
+    case resumeFailed
+
+    fileprivate init(_ nativeError: BillingServiceErrorNative) {
+        switch nativeError.code {
+        case .invalidSkuId:
+            self = .invalidSkuId
+        case .resumeFailed:
+            self = .resumeFailed
+        case .tokenValidationFailed:
+            self = .tokenValidationFailed
+        @unknown default:
+            self = .unknown
+        }
+    }
+}
+
+/// Protocol for `BillingServiceNative` implementation. Inversing the dependency on `BillingServiceNative` allows us
+/// to unit test our implementation.
+protocol BillingService {
+    func getSKUTokenIfValid(for sessionType: BillingHandler.SessionType) -> String
```

Comment:
The protocol won't break. `ProductionBillingService` implementation will break though for obvious reasons. All `BillingService` clients won't break. If `MBXBillingService` will change is a dramatic way, then it would probably require some changes to this protocol as well but we don't expect such changes in foreseeable future.